### PR TITLE
Add partially working HMR for JS/TS/React

### PR DESCRIPTION
**Pull request by Palid**
Please don't ask me questions, I have no idea why it works now and it didn't before, but hey, HMR for react!

Oh, also annoying overlay included. Test this branch out first, I think we should keep it and instead get rid of the errors hitting us in the face.

### CODE REVIEW

**Comment by Palid** [comment]
A bad proof of a good solution.
https://user-images.githubusercontent.com/3636685/128642548-76392714-17cb-4b86-9883-af609ebd0fca.mp4

**Comment by turt2live** [comment]
@Palid why me specifically?

File: package.json [hunk 1]
```diff
@@ -88,6 +88,7 @@
     "@babel/preset-typescript": "^7.12.7",
     "@babel/register": "^7.12.10",
     "@babel/runtime": "^7.12.5",
+    "@pmmmwh/react-refresh-webpack-plugin": "^0.4.3",
```
Comment (dbkr) [inline package.json:91]: hmm - this was last published 2 years ago an is 'experimental'... I guess if it works, it works, but it doesn't exactly inspire confidence.

**Review by dbkr — COMMENTED** [review]
Also, I think the general intention behind auto-merge was for when you're waiting for tests to pass rather than waiting for review. Either way, this does feel like the kind of thing we'd want to merge explicitly in terms of giving people a heads up when it lands.

File: package.json (hunk 1 above)
Comment (Palid) [reply package.json:91]: @dbkr new version is in progress, so don't worry about it. The old HMR was hell to setup in this project and didn't work at all.

**Comment by Palid** [comment]
> @Palid why me specifically?

thought about you as the person hating webpack. 😆

**Comment by turt2live** [comment]
we always request team review: individuals with an interest in something will take a look if they want, but it's not something we directly ask for. Everyone on the team should be able to review any change.

**Comment by Palid** [comment]
> we always request team review: individuals with an interest in something will take a look if they want, but it's not something we directly ask for. Everyone on the team should be able to review any change.

There was both team review and I requested you explicitely, so I think that checks the boxes?

File: package.json (hunk 1 above)
Comment (dbkr) [reply package.json:91]: Oh, I see - they're working on 0.5. Plus I guess 0.4.3 has 2M+ downloads. I guess let's try it then.

**Comment by dbkr** [comment]
I can't approve this yet because you've turned on auto-merge, and this definitely isn't something we want to land immediately before cutting an RC.

**Review by dbkr — APPROVED** [review]

**Comment by SimonBrandner** [comment]
This feels more like `T-Task` to me

**Pull request merged**